runtime/v2/shim.go

```diff
@@ -422,14 +415,18 @@ func (s *shim) CloseIO(ctx context.Context) error {
 }
 
 func (s *shim) Wait(ctx context.Context) (*runtime.Exit, error) {
+	taskPid, err := s.PID(ctx)
```

Comment:
why not just record the taskPid? `s.PID` always send request to shim to get taskPID here.

Reply:
Pid is task service specific thing. The goal of this PR is to make shim stateless and unaware about services shim instance provides. This way we can add new services on top of shims independently - task service, sandbox api, port forwarding, pulling inside sandbox, etc (lots of context described in https://github.com/containerd/containerd/issues/5742). Pid will move to task service back, once we decouple shim v2 `task manager` into `shim manager` + `task service`.

Reply:
ok. thanks for the comment~